addons/hr_recruitment/models/hr_job.py

```diff
@@ -108,6 +109,17 @@ def create(self, vals):
         vals['favorite_user_ids'] = vals.get('favorite_user_ids', []) + [(4, self.env.uid)]
         return super(Job, self).create(vals)
 
+    @api.model
+    def search_panel_select_range(self, field_name):
+        if field_name == 'user_id':
+            users = self.env['hr.job'].search([('user_id', '!=', None)]).mapped('user_id')
+            available_users = self.env['res.users'].search([('id', 'in', users.ids)], limit=SEARCH_PANEL_LIMIT).read(['display_name'])
```

Comment:
This might be even more readable
```suggestion
            users = self.search([('user_id', '!=', False)]).user_id
            available_users = users._filter_access_rules('read')[:SEARCH_PANEL_LIMIT].read(['display_name'])
```